_posts/2018-03-01-spatial-modeling-mlr.markdown

```diff
@@ -0,0 +1,318 @@
+more than 100 modeling packages. In comparison to sperrorest it
+excelaertes by providing the option to easily tune hyperparameters. We
+started with the most common spatial partitioning approach which uses
+k-means clustering (see Brenning ([2012](#ref-sperrorest))). In this
+blog post I will give a practical guide on how to perform a nested
+spatial cross-validation in `mlr`. I will use a SVM as the algorithm
+because it is a widely know and used algorithm that always needs to be
+tuned to achieve good performance. Although the tuning effect of models
+with meaningful default hyperparameter values such as Random Forest is
+not high, one should always conduct a hyperparameter tuning as the
+specific effect is not known a priori.
+
+Everything in `mlr` is build upon the following steps: Create a task,
+(tune hyperparameters), train your model, predict to a new data set.
+When the model performance is to be estimated, this procedure is done
+hundreds of times to reduce the variance among different traing +
+predict arangements. It essentially mimics the situation that you would
+apply your fitted model on 500 different unkown test data sets. You will
+get a different performance on each. However, usually you have no
+in-situ data to check whether your model
+
+Creating a task
+===============
+
+The task essentially stores all information around your data set: Type
+of the response variable, supervised/unsupervised, spatial or
+non-spatial and many more.
+
+I will use the example data set for spatial applications `ecuador` that
+was added to `mlr`. This data set from Jannes Muenchow also comes with
+sperrorest. Since the integration is quite new, we need to use the
+development version of `mlr` from Github.
+
+``` r
+devtools::install_github("mlr-org/mlr")
+
+library("mlr")
+```
+
+First, we need to create a task. Luckily, the built-in data sets of
+`mlr` are already in a task structure. Check sections
+[Task](https://%60mlr%60-org.github.io/%60mlr%60-tutorial/devel/html/task/index.html)
+in the `mlr`-tutorial if you need to build one from scratch. To make it
+a [spatial
+task](https://%60mlr%60-org.github.io/%60mlr%60-tutorial/devel/html/handling_of_spatial_data/index.html),
+you need to provide the coordinates of the data set in argument
+`coordinates` of the task. They will later be used for the spatial
+partitioning of the data set in the spatial cross-validation.
+
+``` r
+data("spatial.task")
+spatial.task
+#> Supervised task: ecuador
+#> Type: classif
+#> Target: slides
+#> Observations: 751
+#> Features:
+#>    numerics     factors     ordered functionals 
+#>          10           0           0           0 
+#> Missings: FALSE
+#> Has weights: FALSE
+#> Has blocking: FALSE
+#> Has coordinates: TRUE
+#> Classes: 2
+#> FALSE  TRUE 
+#>   251   500 
+#> Positive class: TRUE
+```
+
+We are dealing with a supervisd classication problem that has 751
+observations, 10 numeric predicors, the response is “slides” and has a
```

Comment:
predictors

I suggest you do a close proof read yourself, and then ask me to do a review of the second half.

Reply:
Yes. Sorry again, I should have taken a closer look.